app/services/agency_identity_linker.rb

```diff
@@ -26,7 +26,7 @@ def self.for(user:, service_provider:, skip_create:)
     )
 
     uuid = spi&.uuid || SecureRandom.uuid
-    AgencyIdentity.create(
+    AgencyIdentity.find_or_create_by(
```

Comment:
I'm not sure that this is the right method to be changing, the error was in sign up, but this method is only called in an Attempts API class: https://github.com/18F/identity-idp/blob/2fc19c9/app/services/attempts_api/tracker.rb#L98-L102

Reply:
So are you saying to do the call at the point of sign up? Because if we are to do that, we would have to do the same logic that is in the above link, no?
Also, wouldn't this also reduce the chance of a race condition in the Attempts API class? Or does that not matter?

Reply:
Sorry, I was intending to say that the exception occurred during sign up, which calls `find_or_create_agency_identity` further down, and that's probably where any changes would need to occur.

Reply:
My mistake. Changed in [2f448c0](https://github.com/18F/identity-idp/pull/12273/commits/2f448c036ca1d0e90184e64aa41ddf5ac3f7247a).